src/qiskit_qasm3_import/converter.py

```diff
@@ -183,13 +191,23 @@ class ConvertVisitor(QASMVisitor[State]):
     # In some places, such as symbol definitions, we do some simple checks to help everyone's
     # sanity, as the reference package doesn't yet do this.
 
-    # pylint: disable=missing-function-docstring,no-self-use,unused-argument
+    # pylint: disable=missing-function-docstring,no-self-use,unused-argument,protected-access
 
     def convert(self, node: ast.Program, *, source: Optional[str] = None) -> QuantumCircuit:
         """Convert a program node into a :class:`~qiskit.circuit.QuantumCircuit`.  If given,
         `source` is a string containing the OpenQASM 3 source code that was parsed into `node`.
         This is used to generated improved error messages."""
-        return self.visit(node, State(Scope.GLOBAL, source)).circuit
+
+        state = self.visit(node, State(Scope.GLOBAL, source))
+        symbols = state.symbol_table
+        if any(is_physical(name) for name in symbols.keys()):
+            names = filter(is_physical, symbols.keys())
```

Comment:
Very minor, but this is an inefficiency - we iterate through `symbols.keys()` completely twice.  It'd be a shade cleaner to do
```python
physical_indices = [int(name.name[1:]) for name, value in symbols.items() if isinstance(value.type, types.HardwareQubit)]
if physical_indices:
    qr = QuantumRegister(...)
    ...
```

Reply:
Oh yeah. That was one of areas on my refactoring checklist, but it slipped through.